pkg/app/piped/cloudprovider/ecs/client.go

```diff
@@ -0,0 +1,181 @@
+			v.RoleARN = roleARN
+			v.TokenRetriever = stscreds.IdentityTokenFile(tokenPath)
+		}))
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.Background(), optFns...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load config to create ecs client: %w", err)
+	}
+	c.client = ecs.NewFromConfig(cfg)
+
+	return c, nil
+}
+
+func (c *client) CreateService(ctx context.Context, service types.Service) (*ecs.CreateServiceOutput, error) {
+	input := &ecs.CreateServiceInput{
+		ServiceName:                   service.ServiceName,
+		Cluster:                       service.ClusterArn,
+		DeploymentConfiguration:       service.DeploymentConfiguration,
+		DeploymentController:          service.DeploymentController,
+		DesiredCount:                  &service.DesiredCount,
+		EnableECSManagedTags:          service.EnableECSManagedTags,
+		HealthCheckGracePeriodSeconds: service.HealthCheckGracePeriodSeconds,
+		LaunchType:                    service.LaunchType,
+		LoadBalancers:                 service.LoadBalancers,
+		NetworkConfiguration:          service.NetworkConfiguration,
+		PlacementConstraints:          service.PlacementConstraints,
+		PlacementStrategy:             service.PlacementStrategy,
+		PlatformVersion:               service.PlatformVersion,
+		PropagateTags:                 service.PropagateTags,
+		Role:                          service.RoleArn,
+		SchedulingStrategy:            service.SchedulingStrategy,
+		ServiceRegistries:             service.ServiceRegistries,
+		Tags:                          service.Tags,
+		TaskDefinition:                service.TaskDefinition,
+	}
+	output, err := c.client.CreateService(ctx, input)
+	if err != nil {
+		return &ecs.CreateServiceOutput{}, fmt.Errorf("failed to create ECS service %s: %w", *service.ServiceName, err)
+	}
+	return output, nil
+}
+
+func (c *client) UpdateService(ctx context.Context, service types.Service) (*ecs.UpdateServiceOutput, error) {
+	input := &ecs.UpdateServiceInput{
+		Service:                       service.ServiceName,
+		Cluster:                       service.ClusterArn,
+		DeploymentConfiguration:       service.DeploymentConfiguration,
+		DesiredCount:                  &service.DesiredCount,
+		HealthCheckGracePeriodSeconds: service.HealthCheckGracePeriodSeconds,
+		NetworkConfiguration:          service.NetworkConfiguration,
+		PlacementConstraints:          service.PlacementConstraints,
+		PlacementStrategy:             service.PlacementStrategy,
+		PlatformVersion:               service.PlatformVersion,
+		TaskDefinition:                service.TaskDefinition,
+	}
+	output, err := c.client.UpdateService(ctx, input)
+	if err != nil {
+		return &ecs.UpdateServiceOutput{}, fmt.Errorf("failed to update ECS service %s: %w", *service.ServiceName, err)
+	}
+	return output, nil
+}
+
+func (c *client) RegisterTaskDefinition(ctx context.Context, taskDefinition types.TaskDefinition) (*ecs.RegisterTaskDefinitionOutput, error) {
```

Comment:
This definition does not satisfy the Interface that was defined [here](https://github.com/pipe-cd/pipe/pull/1647/files#diff-a65df04aa27280240981bddbcef1d454cab8eff89f8bc858d9959b7029573a76R35).

Reply:
To be simple, let's return `*types.TaskDefinition`  instead of this output.